Fix/login and signup now workin

ryofac opened the pull request:
https://trello.com/c/CDk9miob/9-back-end-auth-adcionar-o-usu%C3%A1rio-no-padr%C3%A3o-de-autentica%C3%A7%C3%A3o-do-djangorest

ryofac committed:
fix - login and signup working

ryofac committed:
feat - implement login and sign-in with class based views

ryofac committed:
feat - add final implement and test validation

HalysonItallo commented on app/social_network/tests.py:
```diff
@@ -1,34 +1,162 @@
-    def test_test_token(self):
-        pass
+    @pytest.mark.django_db
+    def test_signout_is_success(self, api_client: APIClient, created_user: User):
```
Tá faltando os casos de falha dos testes do TestPrivateSocialNetwork

HalysonItallo commented on app/social_network/views.py:
```diff
@@ -1,82 +1,67 @@
-    return Response({"detail": "sucessfuly logged out"}, status=status.HTTP_200_OK)
+class SignoutView(APIView):
+    def post(self, request):
+        if not request.user or not request.user.is_authenticated:
```
Pode trocar essa lógica pela lógica do IsAuthenticated, acho melhor do que ser feito assim

HalysonItallo commented on app/social_network/views.py:
```diff
@@ -1,82 +1,67 @@
-        # Hasheando a senha:
-        created_user.set_password(data["password"])
-        created_user.save()
+        if serializer.is_valid():
```
Tem que colocar quando o serializer não for válido

ryofac committed:
fix - test_sign_out_not_logged_in_fail now expect a 401-unauthorized instead of 400-bad-request

Using the permission and authentication classes makes this process easier, if the user are not logged in, he has no permissions to acess this route.

ryofac committed:
fix - signout view now have the permission and authentication classes
